app/src/test/java/dev/yc/xmrworker/ui/myminer/MinersViewModelTest.kt

```diff
@@ -0,0 +1,180 @@
+package dev.yc.xmrworker.ui.myminer
+
+import dev.yc.xmrworker.InstantExecutorExtension
+import dev.yc.xmrworker.data.ADDRESS
+import dev.yc.xmrworker.data.IDENTIFIER
+import dev.yc.xmrworker.data.repository.FakeSupportXmrRepository
+import dev.yc.xmrworker.ui.myminer.mineritem.MinerState
+import dev.yc.xmrworker.ui.myminer.mineritem.MinerUiState
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(InstantExecutorExtension::class)
+class MinersViewModelTest {
+    private lateinit var viewModel: MinersViewModel
+    private val testDispatcher = UnconfinedTestDispatcher()
+
+    @Test
+    fun fetchMinerData_uiStateSuccess() = runTest {
+        // Arrange
+        val testData = MinerUiState.Success(
+            minerStates = listOf(
+                MinerState(
+                    hash = "7000",
+                    id = IDENTIFIER,
+                    lastTimeInTimestamp = 1657266133,
+                    totalHash = "81130718490",
+                )
+            )
+        )
+        val excepted = listOf(
+            MinerState(
+                hash = "7000",
+                id = IDENTIFIER,
+                lastTimeInTimestamp = 1657266133,
+                totalHash = "81130718490",
+            )
+        )
+        setupViewModel(testData)
+
+        // Action
+        viewModel.fetchMiners()
+
+        // Assert
+        assertEquals(excepted, viewModel.minerData.value)
+    }
+
+    @Test
+    fun fetchMinerDataWithError_errorEventInvoked() = runTest {
+        // Arrange
+        val testData = MinerUiState.Error
+        val excepted = Unit
+        setupViewModel(testData)
+
+        // Action
+        viewModel.fetchMiners()
+
+        // Assert
+        assertEquals(excepted, viewModel.errorEvent.value?.peekContent())
```

Comment:
why we can make sure error event is handled by checking value is Unit, if expected value is Unit, how you can make sure the test result is actually expected(100% correct, won't be impacted by others)